donjuan/edge.py

```diff
@@ -39,31 +38,9 @@ def is_wall(self) -> bool:
             (self.cell1 is None)
             or (self.cell2 is None)
             or (self.cell1.filled != self.cell2.filled)
+            or (
+                self.cell1.space != self.cell2.space
+                and self.cell1.space is not None
+                and self.cell2.space is not None
+            )
```

Comment:
Actually, on second thought, maybe 
`                and self.cell1.space is not None
                and self.cell2.space is not None`
isn't needed. If one was None and the other wasn't, would that always have a wall anyway?

Reply:
Yup.

Reply:
Actually i think the case we are excluding is if both cells are None. In which case it doesn't matter if the edge is a wall or not, but we are asserting "no it isn't". So, I think it doesn't matter either way.